src/parse.rs

```diff
@@ -344,14 +344,31 @@ impl<'a> Bytes<'a> {
         s.bytes().enumerate().all(|(i, b)| self.bytes.get(i).map(|t| *t == b).unwrap_or(false))
     }
 
-    pub fn unsigned_integer<T>(&mut self) -> Result<T> where T: FromStr {
+    pub fn unsigned_integer<T: Num>(&mut self) -> Result<T> {
+        let base = if self.bytes.get(0) == Some(&b'0') {
+            match self.bytes.get(1).cloned() {
```

Comment:
nit: why cloning?

Reply:
So I don't have to match against references.